rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -1217,9 +1219,25 @@ std::string spu_thread::dump_regs() const
 			}
 		}
 
+		auto to_f64 = [](u32 bits)
+		{
+			const u32 abs = bits & 0x7fff'ffff;
+			constexpr u32 mask = (1 << 23) - 1;
+			return std::copysign(abs <= mask ? 0 : std::exp2(static_cast<int>(abs >> 23) - 127) * (1.0 + (abs & mask) / f64{mask + 1}), bits >> 31 ? -1 : 1);
+		};
+
+		const double array[]{to_f64(r.u32r[0]), to_f64(r.u32r[1]), to_f64(r.u32r[2]), to_f64(r.u32r[3])};
+
 		const u32 i3 = r._u32[3];
+		const bool is_packed = v128::from32p(i3) == r;
 
-		if (v128::from32p(i3) == r)
+		if (floats_only)
+		{
+			fmt::append(ret, "%s, %s, %s, %s", array[0], array[1], array[2], array[3]);
```

Comment:
Why %s. Am I mistaking something here?

Reply:
I'll change but it doesnt make much difference.